src/app/services/news.service.spec.ts

```diff
@@ -1,15 +1,679 @@
+          createNewsItem('Test2', true, null, '2018-09-14 18:00:00.000Z'),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(2);
+      }));
+      it('returns a is a higher priority than b',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, '2018-10-14 17:00:00.000Z'),
+          createNewsItem('Test2', true, 2, '2018-09-14 18:00:00.000Z'),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(-1);
+      }));
+      it('returns b is a higher priority than a',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 2, '2018-10-14 17:00:00.000Z'),
+          createNewsItem('Test2', true, 1, '2018-09-14 18:00:00.000Z'),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(1);
+      }));
+    });
+    describe(('given the same priority'), () => {
+      it('returns 0 when a and b are null',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, ''),
+          createNewsItem('Test2', true, 1, ''),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(0);
+      }));
+      it('returns a is more recent than b when b are null',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, '2018-10-14 18:00:00.000Z'),
+          createNewsItem('Test2', true, 1, ''),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(-1539540000000);
+      }));
+      it('returns b is more recent than a when a are null',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, ''),
+          createNewsItem('Test2', true, 1, '2018-10-14 18:00:00.000Z'),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(1539540000000);
+      }));
+      it('returns a is more recent than b',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, '2018-10-14 18:00:00.000Z'),
+          createNewsItem('Test2', true, 1, '2018-10-14 17:00:00.000Z'),
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(-3600000);
+      }));
+      it('returns b is more recent than a',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, '2018-10-14 17:00:00.000Z'),
```

Comment:
try making other values different,not just year.

Reply:
Fixed